File: app/src/test/java/com/myapplication/rick/packagedelivery/TestAddressClass.java
package com.myapplication.rick.packagedelivery;

import org.junit.Test;

/**
 * Created by Rick on 8-5-2016.
 */
public class TestAddressClass {
    private Street unevenStreet = new Street("unevenstreet", new Range(3, 9, RangeType.Uneven));
    private Street evenStreet = new Street("evenstreet", new Range(4, 30, RangeType.Even));

    @Test
    public void AddressCreationUnevenMinimumValue() {
        Address address = new Address(unevenStreet, 3);
    }

    @Test
    public void AddressCreationUnevenMaximumValue() {
        Address address = new Address(unevenStreet, 9);
    }

    @Test (expected = IllegalArgumentException.class)
    public void AddressCreationEvenNumberWithUnevenStreet() throws IllegalArgumentException {
        Address address = new Address(unevenStreet, 4);
    }

    @Test (expected = IllegalArgumentException.class)
    public void AddressCreationUnevenTooHighNumber () throws IllegalArgumentException {
        Address address = new Address(unevenStreet, 11);
    }

    @Test (expected = IllegalArgumentException.class)
    public void AddressCreationUnevenTooLowNumber() throws IllegalArgumentException {
        Address address = new Address(unevenStreet, 1);
    }

    @Test (expected = IllegalArgumentException.class)
    public void AddressCreationUnevenNumberWithEvenStreet() throws IllegalArgumentException {
        Address address = new Address(evenStreet, 9);
    }

    @Test (expected = IllegalArgumentException.class)
    public void AddressCreationNullStreet() throws IllegalArgumentException {
        Address address = new Address(null, 1);
    }
}

File: app/src/test/java/com/myapplication/rick/packagedelivery/TestAddressClass.kt
package com.myapplication.rick.packagedelivery

import org.junit.Test

/**
 * Created by Rick on 8-5-2016.
 */
class TestAddressClass {
    private val unevenStreet = Street("unevenStreet", Range(3, 9, RangeType.Uneven))
    private val evenStreet = Street("evenStreet", Range(4, 30, RangeType.Even))

    @Test
    fun AddressCreationUnevenMinimumValue() {
        val address = Address(unevenStreet, 3)
    }

    @Test
    fun AddressCreationUnevenMaximumValue() {
        val address = Address(unevenStreet, 9)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun AddressCreationEvenNumberWithUnevenStreet() {
        val address = Address(unevenStreet, 4)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun AddressCreationUnevenTooHighNumber() {
        val address = Address(unevenStreet, 11)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun AddressCreationUnevenTooLowNumber() {
        val address = Address(unevenStreet, 1)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun AddressCreationUnevenNumberWithEvenStreet() {
        val address = Address(evenStreet, 9)
    }
}
